lighthouse-core/audits/valid-source-maps.js

```diff
@@ -0,0 +1,162 @@
+  /** Title of a Lighthouse audit that provides detail on HTTP to HTTPS redirects. This descriptive title is shown to users when HTTP traffic is redirected to HTTPS. */
+  title: 'Page has valid source maps',
+  /** Title of a Lighthouse audit that provides detail on HTTP to HTTPS redirects. This descriptive title is shown to users when HTTP traffic is not redirected to HTTPS. */
+  failureTitle: 'Missing source maps for large first-party JavaScript',
+  /** Description of a Lighthouse audit that tells the user that their JavaScript source maps are invalid or missing. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'Source maps translate your minified code to the original source code which ' +
+    'helps with debugging, and tracking down bugs in production. Consider enabling or updating ' +
+    'source maps during JavaScript bundling to take advantage of this feature. ' +
+    '[Learn more](https://developers.google.com/web/tools/chrome-devtools/javascript/source-maps).',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+const LARGE_JS_BYTE_THRESHOLD = 500 * 1024;
+
+/**
+ * Returns true if the script URL is either:
+ * (1) a known third-party script or
+ * (2) the same as the (redirected) requested Lighthouse URL
+ * @param {string} url
+ * @param {string} finalUrl
+ */
+function isFirstParty(url, finalUrl) {
+  try {
+    const entity = thirdPartyWeb.getEntity(url);
+    if (!entity) return true;
+    return entity === thirdPartyWeb.getEntity(finalUrl);
+  } catch (_) {
+    return false;
+  }
+}
+
+class ValidSourceMaps extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'valid-source-maps',
+      title: str_(UIStrings.title),
+      failureTitle: str_(UIStrings.failureTitle),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['ScriptElements', 'SourceMaps', 'URL'],
+    };
+  }
+
+  /**
+   * Returns true if the size of the script exceeds a static threshold
+   * @param {LH.Artifacts.ScriptElement} scriptElement
+   * @param {string} finalURL
+   * @return {boolean}
+   */
+  static isLargeFirstPartyJS(scriptElement, finalURL) {
+    if (scriptElement.content === null) return false;
+
+    const isLargeJS = scriptElement.content.length >= LARGE_JS_BYTE_THRESHOLD;
+    const isFirstPartyJS = scriptElement.src ? isFirstParty(scriptElement.src, finalURL) : false;
+
+    return isLargeJS && isFirstPartyJS;
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} _
+   */
+  static async audit(artifacts, _) {
+    const {SourceMaps} = artifacts;
+
+    /** @type {Set<string>} */
+    const isMissingMapForLargeFirstPartyScriptUrl = new Set();
+
+    let missingMapsForLargeFirstPartyFile = false;
+    const results = [];
+    for (const ScriptElement of artifacts.ScriptElements) {
```

Comment:
```suggestion
    for (const scriptElement of artifacts.ScriptElements) {
```
lowercase camelCase for local vars and properties (for `ScriptElement` and `SourceMap` below).

(It's kind of confusing because `artifacts.Artifact` property names are capitalized, but that's because of an ancient decision to automatically pull the artifact names from the gatherer class names (and class names are capitalized camel case) and it would be way too disruptive to change that now for a little bit more consistency)

Reply:
kk